client/src/annoMatrix/crossfilter.ts

```diff
@@ -240,86 +223,64 @@ export default class AnnoMatrixObsCrossfilter {
-  allSelectedMask() {
+  allSelectedMask(): Uint8Array {
     /* if no data yet indexed in the crossfilter, just say everything is selected */
     if (
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      (this as any).obsCrossfilter.size() === 0 ||
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      (this as any).obsCrossfilter.dimensionNames().length === 0
+      this.obsCrossfilter.size() === 0 ||
+      this.obsCrossfilter.dimensionNames().length === 0
     ) {
       /* fake the mask */
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      return new Uint8Array((this as any).annoMatrix.nObs).fill(1);
+      return new Uint8Array(this.annoMatrix.nObs).fill(1);
     }
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    return (this as any).obsCrossfilter.allSelectedMask();
+    return this.obsCrossfilter.allSelectedMask();
   }
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  allSelectedLabels() {
+  allSelectedLabels(): Int32Array {
     /* if no data yet indexed in the crossfilter, just say everything is selected */
     if (
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      (this as any).obsCrossfilter.size() === 0 ||
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      (this as any).obsCrossfilter.dimensionNames().length === 0
+      this.obsCrossfilter.size() === 0 ||
+      this.obsCrossfilter.dimensionNames().length === 0
     ) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      return (this as any).annoMatrix.rowIndex.labels();
+      return this.annoMatrix.rowIndex.labels();
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    const mask = (this as any).obsCrossfilter.allSelectedMask();
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    const index = (this as any).annoMatrix.rowIndex.isubsetMask(mask);
+    const mask = this.obsCrossfilter.allSelectedMask();
+    const index = this.annoMatrix.rowIndex.isubsetMask(mask);
     return index.labels();
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'array' implicitly has an 'any' type.
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  fillByIsSelected(array, selectedValue, deselectedValue) {
+  fillByIsSelected(
+    array: Float32Array,
```

Comment:
`array` param and return value are any array or typedarray type.

Reply:
Updated [crossfilter.ts::fillByIsSelected](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/crossfilter.ts#L276).

Bruce, should the `array` param and return value be constrained here to be the same? If so, could you suggest a way to constrain them?

Reply:
Suggest:

```
fillByIsSelected<T>(array: T, selectedValue: T[0], deselectedValue: T[0]): T {
   ...
}
```

This uses indexed access types: https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html

Reply:
I'm not sure if I have got this one right; I used a [slightly different syntax](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/crossfilter.ts#L286) and added one corresponding [suppressed compile error](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/crossfilter.ts#L296).

Reply:
Yeah, I gave you a slightly wrong suggestion.  If you do the following, it should work and will be compatible with my upcoming crossfilter PR:

```
  fillBySelection<A extends TypedArray>(
    result: A,
    selectedValue: A[0],
    deselectedValue: A[0]
  ): A { ... } 
``` 

Sorry about that!   AnyArray screws up stuff because it contains `unknown[]`

Reply:
My apologies if I'm missing something here. I have updated `fillBySelection` to use [TypedArray](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/crossfilter.ts#L286) but am still requiring a [suppression](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/crossfilter.ts#L296)).

Reply:
Mmmm...likely because I have not yet landed the crossfilter PR (it _will_ be compatible, but is not yet).

I suggest leaving in the suppression (with this change), and let me clean it up as I do the crossfilter PR (forthcoming soon)

Reply:
Got it, thanks Bruce!